src/main/java/it/pagopa/transactions/commands/handlers/TransactionRequestAuthorizationHandlerCommon.java

```diff
@@ -117,22 +117,30 @@ private Mono<AuthorizationOutput> walletNpgCardsPaymentFlow(
                                                                 String lang,
                                                                 UUID userId
     ) {
-        return paymentGatewayClient
-                .requestNpgBuildSession(authorizationData, correlationId, true, clientId, lang, userId)
-                .flatMap(
-                        orderIdAndFieldsDto -> transactionTemplateWrapper.save(
-                                new TransactionCacheInfo(
-                                        authorizationData.transactionId(),
-                                        new WalletPaymentInfo(
-                                                orderIdAndFieldsDto.getT2().getSessionId(),
-                                                orderIdAndFieldsDto.getT2().getSecurityToken(),
-                                                orderIdAndFieldsDto.getT1()
+        return Mono.justOrEmpty(authorizationData.contextualOnboardDetails())
+                .map(details -> Tuples.of(details.orderId(), authorizationData.sessionId().orElseThrow()))
+                .switchIfEmpty(
+                        Mono.defer(
+                                () -> paymentGatewayClient
```

Comment:
```suggestion
                        paymentGatewayClient
```
maybe we can get rid of the mono defer here since the paymentGatewayClient.requestNpgBuildSession code already returns a mono.
what do you think?